dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/query/JpaCriteriaQueryEngine.java

```diff
@@ -108,68 +111,11 @@ public List<T> query(Query query) {
     CriteriaQuery<T> criteriaQuery = builder.createQuery(klass);
     Root<T> root = criteriaQuery.from(klass);
 
-    if (query.isEmpty()) {
```

Comment:
FYI: This path was eliminated as it should be equal to the "else" path except for the starting point being an empty conjunction.

Note that this path did not add orders because `isEmpty` is the absence of filters and orders. In the refactored code the effect is the same but the orders are always added when non-empty.